# Instagram Likes Report
**Melyssa Calegari Rein**

09/15/2023

I have had my Instagram account since 2012, which means I was around 9 years old at the time I created it. As many people, when creating an Instagram account the first people I followed was my family members and friends. However, as time went by, I started following famous people, business accounts, and more. 

Throughout these last 11 years, I believe that among the top 20 most liked accoounts, I liked more celebrity posts than family and friend's posts. Even though I follow family and friends for a longer time, I do believe I don't like their content as much because they to post less than celebrities. The majority of he famous people I follow post almost every single day, if not more than once a day. In the meantime, I have some friends that post constantly every week or so, but many of them don't post often. So, my hypothesis is that I like celebrities content more than my family/friends' content. 

The data used to this analysis was collected by Instagram and I got access to it by requesting and downloading it as json files from the platform itself. Instagram collects this type of data, from what people post to who people follow and interact with, so they can understand and learn more about the users in order to send them recommendations and ads that better fit their profiles. This way, the users are more easily alienaded to the platform, as what they are receiving is based on what Instagram understood as their likes and interests. 

As all the information gathered was provided by me posting, using Instagram on my devices, and interacting with other accounts, it is reliable in the sense that I know that everything in this report are facts that come from my actions. However, I am also aware that Instagram only selects the data that they want to collect by their importance as something that can be used by them to understand the user better, and that they choose the ones they want to share with us, users. In addition, there is no actual method to confirm if every single information is a hundred percent correct, as they are data collected by Instagram.

## Initial process
After downloading the files from Instagram, I imported pandas to this notebook file and imported json, so that I could load the liked_post.json files - which is the specific file for liked posts and reels inside the data files Instagram provided - and give it a name to call it more easily.

In [12]:
import pandas as pd
import json

In [13]:
with open("/Users/mel/Downloads/melyssacalegari_20230904_part_2/likes/liked_posts.json") as lp:
    liked_posts=json.load(lp)

## Analysis
To start the analysis, I transformed the document into a data frame, which is putting the information of the file into a table. Then, added the timestamp from each like to the table, as it does not appear on it at first due to the way Instagram decided to gather and store this data.

It is important to note that 'likes_media_likes' is the name of the key present inside the dictionary in the json file for liked posts. In addition, that 'timestamp' is inside of 'string_list_data', which is inside of 'likes_media_likes'.

In [14]:
liked_acc=pd.DataFrame( liked_posts['likes_media_likes'])

In [15]:
time_list = [x['string_list_data'][0]['timestamp'] for x in liked_posts['likes_media_likes']]
time_list

[1693792008,
 1693791916,
 1693658196,
 1693652199,
 1693623852,
 1693600235,
 1693589273,
 1693529393,
 1693529390,
 1693529280,
 1693483761,
 1693451496,
 1693451388,
 1693451380,
 1693451319,
 1693450556,
 1693444457,
 1693444151,
 1693444123,
 1693424298,
 1693356989,
 1693356897,
 1693356819,
 1693356790,
 1693356781,
 1693356723,
 1693356506,
 1693356503,
 1693356500,
 1693356489,
 1693356470,
 1693356440,
 1693356426,
 1693313747,
 1693310410,
 1693310287,
 1693282337,
 1693282298,
 1693277716,
 1693251655,
 1693190059,
 1693178905,
 1693178792,
 1693178783,
 1693178775,
 1693178765,
 1693178757,
 1693178738,
 1693178730,
 1693178702,
 1693178698,
 1693178696,
 1693178662,
 1693178654,
 1693178650,
 1693178644,
 1693178639,
 1693178634,
 1693178622,
 1693178618,
 1693178611,
 1693178609,
 1693178603,
 1693178599,
 1693178595,
 1693178593,
 1693178587,
 1693178581,
 1693178579,
 1693178568,
 1693178565,
 1693178048,
 1693177277,
 1693177268,
 1693177263,
 1693173899,
 1693161905,

In [16]:
liked_acc['timestamp']=time_list

The line of code above is responsible for adding a collumn called 'timestamp' to the data frame.

In [30]:
liked_acc.tail()

title                                   string_list_data  \
11683  analuciaprzs  [{'href': 'https://www.instagram.com/p/gvK8yKk...   
11684    evelisercm  [{'href': 'https://www.instagram.com/p/gvlusqx...   
11685  analuciaprzs  [{'href': 'https://www.instagram.com/p/goBPkME...   
11686  analuciaprzs  [{'href': 'https://www.instagram.com/p/glcT2Tk...   
11687  analuciaprzs  [{'href': 'https://www.instagram.com/p/glc_ijk...   

        timestamp  
11683  1384552224  
11684  1384552220  
11685  1384286611  
11686  1384279630  
11687  1384279624

In [29]:
liked_acc.head()

title                                   string_list_data  \
0   luanajohnson._  [{'href': 'https://www.instagram.com/reel/Cwu-...   
1         gnabnahc  [{'href': 'https://www.instagram.com/p/CwvE3GW...   
2  ateez_official_  [{'href': 'https://www.instagram.com/p/CwsEFWi...   
3   this_is_baekho  [{'href': 'https://www.instagram.com/reel/CwrU...   
4   larissamanoela  [{'href': 'https://www.instagram.com/p/Cwp3qdc...   

    timestamp  
0  1693792008  
1  1693791916  
2  1693658196  
3  1693652199  
4  1693623852

These summarized tables above show now the position of each like, the account the like was directed to, the link to it along with the emoji used, and the time it was liked (in milliseconds). The first data frame shows the oldest likes, while the second data frame shows the most recent ones.

Looking at the data frames above, I noted that I was correct about the first couple of liked users of when I created my account being of acquaintances as "analuciaprzs" is my mom's friend and "evelisercm" is my aunt. It is also notable that the last couple of likes were directed to celebrities accounts, apart from the last one ("luanajohnson._") that is a friend's account.

After getting the data frame with each account and its correspondent timestamp, I proceeded to group the likes by account in order to have the exact amount of times I liked each account's posts. I ordered the table from most liked to least liked, as my goal is to discover if I like more celebrities' accounts or family/friends' accounts. Because I want to analyze my top 20 likes, I decided to only look at them, so I shortened the complete grouped table to show only the 20 most liked accounts. 

I also decided to drop the timestamp column for this next data frame because it was giving the same values as the "string_list_data" column. This happens because it is counting how many different timestamps there was per account, which is basically the same amount of different likes per account. 

In [21]:
most_liked = liked_acc.groupby('title').count().drop(columns='timestamp').sort_values('string_list_data',ascending=False).head(20)
most_liked

string_list_data
title                                 
larissamanoela                     534
realstraykids                      463
ateez_official_                    253
1.d.memes                          245
shadowhunterstv                    239
mileycyrus                         227
official_theboyz                   224
vitoriacalegari                    205
imersaemlivros                     201
mefazrir                           161
kat.mcnamara                       135
kayceerice                         119
smart.ib.memes                      96
vecapoccia                          87
enhypen                             81
miraculouspowersbr                  80
nowunited                           78
topstretching.app                   75
tudobeen                            74
anygabriellyofficial                68

### Accounts

From this list, the following are my family members and friends's accounts:
- vitoriacalegari: she is my aunt;
- vecapoccia: she is a family friend, and is the daugther of my sister's godmother.

The account "imersaemlivros" is accually my second account dedicated to books, so I will not be considering it as neither family or celebrity. Other accounts will not count as neither category as well. They are "1.d.memes," "shadowhunterstv," "mefazrir," "smart.ib.memes," "miraculouspowersbr," "topstretching.app." These users are meme accounts or accounts dedicated to TV shows, which I am not considering celebrity account -- as they are not a famous person or group.

The celebirites accounts on the table above are:

#### larissamanoela
![](https://www.diariodepernambuco.com.br/static/app/noticia_127983242361/2023/08/14/936372/20230814152921448524i.webp)
Larissa Manoela is a Brazilian actress and singer that has a long career, as she has been working since she was a child. She might be my top liked because I follow her for a long time to know more about her projects.

#### realstraykids
![](https://assets.teenvogue.com/photos/612f8bf4e8c260b3d41cc82c/1:1/w_2000,h_2000,c_limit/Stray%20Kids%20(1).jpg)
Stray Kids is a k-pop group and although I only started following them in the past two years, I am not surprised to see them on top 2 since I normally go after their posts a lot. 

####  ateez_official_
![](https://www.envimedia.co/wp-content/uploads/2022/07/ateez-1.jpeg)
Ateez is also a k-pop group that I have followed for the past two years but I don't keep a look out for their new posts as much as I do with Stray Kids; I normally like their content when they appear on my feed, which seems to be quite a lot since in the past 2 years I liked 253 of the their posts. However, this group, differently from the next ones, I do keep an eye for new projects, so sometimes I look them up on Instagram to know more of what they have been doing.

####  official_theboyz
![](https://cdn.i-scmp.com/sites/default/files/d8/images/methode/2020/10/20/73ebbbb2-0a09-11eb-afc8-92e0da0ef1c3_image_hires_113807.jpeg)
The Boyz is a k-pop group that I follow for nearly 2 years as well, but I only interact with their posts if they appear on my feed when I am scrolling through it. 

####  kat.mcnamara 
![](https://mn2s-content.s3.eu-west-2.amazonaws.com/wp-content/uploads/2019/02/19171932/kat-mcnamara-mn2s-2-.png)
Kat McNamara is an actress that had the main role on a TV adaptation of my favorite book series of 2015. Because of this, I ended up following her and interacting a lot with her content to know more about the cast and the TV show. I am surprised that my interactions with her posts from 2016 to 2019 were so much that put her at 11th place of most liked accounts in a report of 11 years.

####  kayceerice
![](https://pbs.twimg.com/media/EUEGJDOWoAQXi10.jpg)
Kaycee Rice is a dancer that I have been following for years now, but I interacted more with her posts years ago, although I still keep an eye on her projects as I am a huge fan of her work and talent.

####  enhypen
![](https://assets.teenvogue.com/photos/5fd6e2f4e6ff71e902f97c4e/16:9/w_2560%2Cc_limit/ENHYPEN_DUSK_Photo%2520Courtesy%2520BELIFT%2520LAB_1.jpg)
Enhypen is another k-pop group that I follow for almost 2 years now and I interact more with their content if it appears on my feed.

####  nowunited
![](https://tempersonalizados.com.br/wp-content/uploads/2022/04/Now-United_-conheca-as-tendencias-trazidas-pelo-grupo.jpg)
Now United is a group of dancers and singers that I used to engage with their content more around 2019, but I do like their content from time to time if it appears on my feed. But I am also surprised that I liked their content enough to appear on my top 20.

####  anygabriellyofficial
![](https://s2.glbimg.com/yP7OPJrb37Fkso6NPyoaFknUKzM=/top/e.glbimg.com/og/ed/f/original/2020/09/01/any-gabrielly-8.jpg)
Any Gabrielly is a Brazilian singer and dancer that used to be on Now United but now she is a soloist. Just as Now United, I used to interact more with her content around 2019-2020, but I do engage with it a little bit more nowadays compared to Now United.

So, if we add the amount of times I liked my family's posts and a celebrity's posts from this list, we have the following results:

In [22]:
family = 205 +  86
family

291

In [24]:
celebrities = 534 + 463 + 253 + 227 +224 + 135 +119 + 81 + 78 + 68
celebrities

2182

The following equation if to discover the actual ratio between each category:

In [28]:
celebrities / family

7.498281786941581

# Conclusion
My hypothesis of me liking more celebrities than family and friends' content was correct, as the ratio between them is nearly 7.5:1.

The limitations of using only the liked_posts for this analysis is that it only focuses on reels and posts, not on stories. Hence, for a more specific result on which type of accounts and specific users I normally like and engage more would be interesting to analize the story_likes.json file as well, which has data of every story I liked. Not only the users on top 20 could change completely with this added information, but the ratio between family/friends and celebrities posts could change by a lot. Plus, as I have an additional account in which I follow authors, that in my perspective are considered celebrities if they are famous ones, it would also be of value to this analysis if I got the data for this other account to use it in the report. 

It is also necessary to take into account that there were years that I barely used this Instagram account and that some of them I used more my other account, which can have an influence on the results. In addition, reels was only introduced in the past few years, which means that is a difference between past and present state of the platform important to have in mind.

Another way of testing my hypothesis would be creating a list with the people I consider my family and friends and another list with people who are celebrities, generating a code that could calculate the result so I do not need to make the calculation by myself.

Thus, the next steps for further investigating this hypothesis would be getting data on my additional account along with using story_likes.json to have a broader idea of my most liked accounts. Plus, creating a code that could detect each user and fit them in their respective category, so the calculation could be done with code as well. With such code instructions, it would be even possible to expand the hypothesis to not only inside my top 20 most liked accounts, but to every account I liked. And the hypothesis could be even narrowed down to the ratio I believe it will have between them including this other data.